File: super-digit.py
def superDigit(n, k):
    number = n
    repetitions = k
    
    initial_sum = sum(int(digit) for digit in number) * repetitions
    
    while initial_sum > 9:
        initial_sum = sum(int(digit) for digit in str(initial_sum))

    return initial_sum

File: super-digit.js
function superDigit(n, k) {
    
    let initialSum = n.split('').reduce((sum, digit) => sum + parseInt(digit), 0) * k;
    
    while (initialSum > 9) {
    
        initialSum = initialSum
            .toString()
            .split('')
            .reduce((sum, digit) => sum + parseInt(digit), 0);
    }

    return initialSum
}
